source/server/http/admin.cc

```diff
@@ -530,18 +531,23 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
   Http::Code rc = Http::Code::OK;
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
 
-  const bool show_all = params.find("usedonly") == params.end();
+  const bool used_only = params.find("usedonly") != params.end();
   const bool has_format = !(params.find("format") == params.end());
+  const bool has_regex = params.find("filter") != params.end();
+  const absl::optional<std::regex> regex =
+      has_regex ? absl::optional<std::regex>{std::regex(params.at("filter"))} : absl::nullopt;
 
   std::map<std::string, uint64_t> all_stats;
   for (const Stats::CounterSharedPtr& counter : server_.stats().counters()) {
-    if (show_all || counter->used()) {
+    if ((!used_only || counter->used()) &&
```

Comment:
nit: can we move this condition checks to a function that takes `Metric`  as argument along with other required ones, so that it can be reused for all three types?

Reply:
I like this -- I'll add a function to encapsulate this, since it appears 4x.

Reply:
See 4efc425.